package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -552,16 +582,46 @@ def _read_next_timestep(self):
 
         indices = np.zeros(self.n_atoms, dtype=int)
 
-        f.readline()  # ITEM ATOMS etc
-        for i in range(self.n_atoms):
-            idx, _, xs, ys, zs = f.readline().split()
+        atomline = f.readline()  # ITEM ATOMS etc
+        attrs = atomline.split()[2:]  # attributes on coordinate line
+        col_ids = {attr: i for i, attr in enumerate(attrs)}  # column numbers
+        keys = col_ids.keys()
+        # check for ids and what type of coordinate convention
+        ids = "id" in keys
+        # coord_dict keys = convention keys = column data or False
+        coord_dict = {conv: False for conv in self._conventions[1:]}
+        for coordtype, col_names in self._coordtype_column_names.items():
+            if all(axis in col_ids for axis in col_names):
+                coord_dict[coordtype] = [col_ids[axis] for axis in col_names]
+
+        # this should only trigger on first read of "ATOM" card, after which it
+        # is fixed to the guessed value. Auto proceeds unscaled -> scaled
+        # -> unwrapped -> scaled_unwrapped
+        if self.lammps_coordinate_convention == "auto":
+            match = [var for var in self._conventions[1:] if coord_dict[var]]
+            try:
+                self.lammps_coordinate_convention = match[0]
+            except IndexError:
+                raise ValueError("no coordinate information detected")
+
+        if not coord_dict[self.lammps_coordinate_convention]:
+            raise ValueError(
+                f"no coordinate information of type"
+                f"{self.lammps_coordinate_convention} present in timestep")
+        else:
+            coord_cols = coord_dict[self.lammps_coordinate_convention]
 
-            indices[i] = idx
-            ts.positions[i] = xs, ys, zs
+        for i in range(self.n_atoms):
+            fields = f.readline().split()
+            if ids:
+                indices[i] = fields[col_ids["id"]]
```

Comment:
This seems to be the only use of the `ids` variable, and it's far enough below that it's easy to forget what `ids` is by the time you get here. It's also a little convoluted in checking. Could you just not use the `ids` variable?

```python
try:
    indices[i] = attr_to_col_ix["id"]
except KeyError:
    pass
```

or 

```python
if "id" in col_ids:
   indices[i] = fields[col_ids["id"]]
```

Reply:
I'd probably vote for setting id just above the `for` loop? (just to avoid the need to do lots of `KeyError`s if it's missing or doing the extra `in` operation for natom times.